stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/ChatInfoItem.kt

```diff
@@ -10,92 +10,45 @@ sealed class ChatInfoItem {
-        abstract val textResId: Int
-
-        @get:ColorRes
-        abstract val tintResId: Int
-
-        open val showRightArrow: Boolean = true
-
-        object SharedMedia : Option() {
-            override val iconResId: Int
-                get() = R.drawable.ic_media
-            override val textResId: Int
-                get() = R.string.chat_info_option_media
-            override val tintResId: Int
-                get() = R.color.black
-        }
-
-        object SharedFiles : Option() {
-            override val iconResId: Int
-                get() = R.drawable.ic_files
-            override val textResId: Int
-                get() = R.string.chat_info_option_files
-            override val tintResId: Int
-                get() = R.color.black
-        }
-
-        object SharedGroups : Option() {
-            override val iconResId: Int
-                get() = R.drawable.ic_new_group
-            override val textResId: Int
-                get() = R.string.chat_info_option_shared_groups
-            override val tintResId: Int
-                get() = R.color.black
-        }
-
-        object DeleteConversation : Option() {
-            override val iconResId: Int
-                get() = R.drawable.ic_delete_contact
-            override val textResId: Int
-                get() = R.string.chat_info_option_delete_conversation
-            override val tintResId: Int
-                get() = R.color.red
-            override val showRightArrow: Boolean = false
-        }
-
-        sealed class Stateful : Option() {
-            abstract val isChecked: Boolean
-
-            data class Notifications(override val isChecked: Boolean) : Stateful() {
-                override val iconResId: Int
-                    get() = R.drawable.ic_notifications
-                override val textResId: Int
-                    get() = R.string.chat_info_option_notifications
-                override val tintResId: Int
-                    get() = R.color.black
-            }
-
-            data class Mute(override val isChecked: Boolean) : Stateful() {
-                override val iconResId: Int
-                    get() = R.drawable.ic_mute
-                override val textResId: Int
-                    get() = R.string.chat_info_option_mute_user
-                override val tintResId: Int
-                    get() = R.color.black
-            }
-
-            data class Block(override val isChecked: Boolean) : Stateful() {
-                override val iconResId: Int
-                    get() = R.drawable.ic_block
-                override val textResId: Int
-                    get() = R.string.chat_info_option_block_user
-                override val tintResId: Int
-                    get() = R.color.black
-            }
-        }
-    }
+enum class OptionType(
+    @DrawableRes val iconResId: Int,
+    @StringRes val textResId: Int,
+    @ColorRes val tintResId: Int,
+    val showRightArrow: Boolean = true
```

Comment:
This property only seems to make sense for non-stateful options, I think this modelling could be improved somehow, more clearly separating those two large groups.

What was the issue with the sealed class based approach?

Reply:
You're right. IMO the approach with enum classes is more readable but it introduces some unnecessary parameters. I'll revert to the old version